flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/EvictingWindowOperator.java

```diff
@@ -179,7 +179,7 @@ public void merge(W mergeResult,
 						// if we have no state, there is nothing to do
 						continue;
 					}
-					emitWindowContents(actualWindow, contents, evictingWindowState);
+					emitWindowContents(actualWindow, contents, evictingWindowState, triggerResult.isPurge());
 				}
 
 				if (triggerResult.isPurge()) {
```

Comment:
Does the `if (triggerResult.isPurge())` still needed?  Could we do both the emit and purge  operation in the new function

Reply:
Good idea. Merge emit and purge in the function `emitWindowContents ` and rename it as `triggerWindowContents`.